### ISSUE TITLE
Media merging adjacent

### ISSUE THREAD

**Issue by sarguvlad** [jakubpawlowicz/clean-css#709]
Hi there, first of all I would like to thank you very much for this awesome tool. It saves me a lot of time and it definitely makes the web better and faster.

So here's my issue. I'm running a Windows 8.1, Clean-css 3.4.8, through the gulp-minify-css. I've passed only the 'mediaMerging: true' option.

I have many 'less' modules (files) compiled into one single CSS file and I run gulp-minify-css on it and looks like it gonna merge only the closest media and leave intact the other ones. As you may expect, I have a less file for each module and it would be great to write media queries in each module file instead of having one big file that contains in a mess all the media queries for all the modules. Any help would be highly appreciated!

**style.less**
```less
@import 'foo.less';
@import 'bar.less';
```

**foo.less**
```less
.foo-class{
	background:red;
}

.bar-class{
	background: green;
}

@media (max-width:768px) {
    .foo-class {
        background-color: green;
    }
}

@media (max-width:768px) {
    .bar-class {
        background-color: blue;
    }
}
```
**bar.less**
```less
.foo-bar-class{
	background: red;
}

.bar-foo-class{
	background: green;
}

@media (max-width:768px) {
    .foo-bar-class {
        background-color: orange;
    }
}
```
**output**
```css
.foo-class{
	background:red;
}

.bar-class{
	background: green;
}

@media (max-width:768px) {
    .foo-class {
        background-color: green;
    }

    .bar-class {
        background-color: blue;
    }
}

.foo-bar-class{
	background: red;
}

.bar-foo-class{
	background: green;
}

@media (max-width:768px) {
    .foo-bar-class {
        background-color: orange;
    }
}
```
**expected output**
```css
.foo-class{
	background:red;
}

.bar-class{
	background: green;
}

.foo-bar-class{
	background: red;
}

.bar-foo-class{
	background: green;
}

@media (max-width:768px) {
	.foo-class {
	    background-color: green;
	}

	.bar-class {
	    background-color: blue;
	}

    .foo-bar-class {
        background-color: orange;
    }
}
```

**Comment by ngyikp** [comment]
See also: https://github.com/jakubpawlowicz/clean-css/issues/687

**Comment by sarguvlad** [comment]
Thank you @ngyikp. Does that mean it's not implemented yet? I thought it's working since v1.4.

**Comment by jakubpawlowicz** [comment]
Thanks @ngyikp - it's closely related to #687 indeed. Please note in case of class rules, where many can be applied to same element, it may not be safe to change order. The current implementation is safe but #687 will address id selectors.

In the meantime closing this one as a duplicate.
